### ISSUE TITLE
Assert fails on epoch_state post_propose

### ISSUE THREAD

**Issue by overvenus** [tikv/tikv#8781]
## Bug Report

https://github.com/tikv/tikv/blob/4a828f036fb1c92d2d5e4341089e77edf858bfde/components/raftstore/src/store/peer.rs#L251-L257

```
[2020/10/02 14:19:33.772 +08:00] [FATAL] [lib.rs:481] ["assertion failed: self.last_conflict_index(epoch_state.check_ver,
                         epoch_state.check_conf_ver).is_none()"] [backtrace="stack backtrace:
   0: tikv_util::set_panic_hook::{{closure}}
             at components/tikv_util/src/lib.rs:480
   1: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:524
   2: std::panicking::begin_panic
             at /home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panicking.rs:450
   3: raftstore::store::peer::CmdEpochChecker<S>::post_propose
             at /home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/macros.rs:13
   4: raftstore::store::peer::Peer<EK,ER>::propose
             at /home/stn/tikv/components/raftstore/src/store/peer.rs:2080
      raftstore::store::fsm::peer::PeerFsmDelegate<EK,ER,T,C>::propose_raft_command
             at /home/stn/tikv/components/raftstore/src/store/fsm/peer.rs:3119
   5: raftstore::store::fsm::peer::PeerFsmDelegate<EK,ER,T,C>::handle_msgs
             at /home/stn/tikv/components/raftstore/src/store/fsm/peer.rs:459
   6: <raftstore::store::fsm::store::RaftPoller<EK,ER,T,C> as batch_system::batch::PollHandler<raftstore::store::fsm::peer::PeerFsm<EK,ER>,raftstore::store::fsm::store::StoreFsm<EK>>>::handle_normal
             at /home/stn/tikv/components/raftstore/src/store/fsm/store.rs:865
   7: batch_system::batch::Poller<N,C,Handler>::poll
             at /home/stn/tikv/components/batch-system/src/batch.rs:324
   8: batch_system::batch::BatchSystem<N,C>::spawn::{{closure}}
             at /home/stn/tikv/components/batch-system/src/batch.rs:396
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/sys_common/backtrace.rs:130
   9: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at /home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/mod.rs:475
      <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panic.rs:318
      std::panicking::try::do_call
             at /home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panicking.rs:342
      std::panicking::try
             at /home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panicking.rs:319
      std::panic::catch_unwind
             at /home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/panic.rs:394
      std::thread::Builder::spawn_unchecked::{{closure}}
             at /home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/thread/mod.rs:474
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/ops/function.rs:233
  10: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/16957bd4d3a5377263f76ed74c572aad8e4b7e59/src/liballoc/boxed.rs:1078
      <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
             at /rustc/16957bd4d3a5377263f76ed74c572aad8e4b7e59/src/liballoc/boxed.rs:1078
      std::sys::unix::thread::Thread::new::thread_start
             at src/libstd/sys/unix/thread.rs:87
  11: start_thread
  12: __clone
"] [location=/home/stn/.rustup/toolchains/nightly-2020-07-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/macros.rs:13] [thread_name=raftstore-8-3]
```

### What version of TiKV are you using?

master

[epoch_state.tar.gz](https://github.com/tikv/tikv/files/5316445/epoch_state.tar.gz)

**Comment by gengliqi** [comment]
It is introduced by https://github.com/tikv/tikv/pull/8441 and it is fixed in https://github.com/tikv/tikv/pull/8499/commits/2c478cde2f2a30e474310cfbafe381a0f396170f with https://github.com/tikv/tikv/pull/8499.

**Comment by youjiali1995** [comment]
Fixed by #8499.
